### PR TITLE
4758: hasVerifiedIrsNotice bug fixes

### PR DESCRIPTION
**Pull request by rachaelparris**
Also removed `hasIrsNotice` from the Case entity so that it is no longer stored. It's not used for anything other than showing/hiding the IRS notice date in the UI for a petitioner creating a case.

### CODE REVIEW

File: shared/src/business/entities/cases/CaseInternal.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ function CaseInternal(rawCase) {
   this.caseCaption = rawCase.caseCaption;
   this.caseType = rawCase.caseType;
   this.filingType = rawCase.filingType;
+  this.irsNoticeDate = rawCase.irsNoticeDate;
+  this.hasVerifiedIrsNotice = rawCase.hasVerifiedIrsNotice || false;
```
Comment (matthopson) [inline shared/src/business/entities/cases/CaseInternal.js:31]: If this is optional, does setting a default value have any adverse affects anywhere else?

File: shared/src/business/entities/cases/CaseInternal.js [hunk 2]
```diff
@@ -110,6 +112,8 @@ const paperRequirements = joi
     ),
     caseCaption: joi.string().required(),
     caseType: joi.string().required(),
+    hasVerifiedIrsNotice: joi.boolean().required(),
```
Comment (matthopson) [inline shared/src/business/entities/cases/CaseInternal.js:115]: Oh, I see, its required here.

File: shared/src/business/utilities/getFormattedCaseDetail.js [hunk 3]
```diff
@@ -303,11 +303,7 @@ const formatCase = (applicationContext, caseDetail) => {
 
   result.datePetitionSentToIrsMessage = result.irsDateFormatted;
 
-  result.shouldShowIrsNoticeDate =
-    result.hasVerifiedIrsNotice ||
-    ((result.hasVerifiedIrsNotice === null ||
-      result.hasVerifiedIrsNotice === undefined) &&
-      result.hasIrsNotice);
+  result.shouldShowIrsNoticeDate = result.hasVerifiedIrsNotice;
```
Comment (matthopson) [inline shared/src/business/utilities/getFormattedCaseDetail.js:306]: 68747470733a2f2f36362e6d656469612e74756d626c722e636f6d2f35656463616232373163303433613435626638383532386639383434396530642f74756d626c725f7038687133625a35704c317331763372316f325f3530302e676966

**Review by matthopson — APPROVED** [review]

File: shared/src/business/entities/cases/CaseInternal.js (hunk 1 above)
Comment (rachaelparris) [reply shared/src/business/entities/cases/CaseInternal.js:31]: It's defaulted to No in the UI, so I defaulted to false here. This should only be set when a petitions clerk is creating a paper case - that's the only time this entity is used and the value is not defaulted in the regular Case entity.

**Pull request merged**